### PR TITLE
54 add credit usage dashboard for users

### PR DESCRIPTION
**Pull request by jcreek**
## Release Notes

- **New Features**
	- Introduced a credit dashboard component displaying visualisations of credit data and transaction history.
	- Added functionality to fetch and display user credit transactions and remaining credits.
	- Implemented monthly aggregates for credits added and deducted.
	- New dependencies added for enhanced data visualisation capabilities, including `chart.js` and `simple-datatables`.
	- Added a "Credit Usage" section in the account page for monitoring credit balance.

- **Bug Fixes**
	- Enhanced error handling for user authentication and data retrieval.

- **Documentation**
	- Updated comments to clarify the purpose and parameters of new functions.

### CODE REVIEW

File: src/routes/(user)/credits/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+		padding: 0.5rem 1rem;
+		border: 1px solid #ccc;
+		border-radius: 0.5rem;
+		cursor: pointer;
+	}
+	button.font-bold {
+		font-weight: bold;
+		background-color: #f0f0f0;
+	}
+</style>
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/routes/(user)/credits/+page.svelte:163]: Good point, but not necessary for the MVP. Please create an issue for this for milestone 2.0.0

File: src/routes/(user)/credits/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+					{
+						label: 'Cumulative Balance',
+						data: cumulativeBalanceData,
+						borderColor: 'blue',
+						fill: false
+					}
+				]
+			};
+		}
+	}
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/routes/(user)/credits/+page.svelte:69]: Not needed for MVP, create a ticket to do this in milestone 2.0.0

File: src/routes/(user)/credits/+page.server.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	try {
+		const creditsRemaining = await getUserCredits(userId);
+		const transactions = await getUserCreditTransactions(userId);
+
+		const monthlyAggregates = transactions.reduce((acc, transaction) => {
+			const date = new Date(transaction.created_at);
+			const month = `${date.getFullYear()}-${date.getMonth() + 1}`;
+
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/routes/(user)/credits/+page.server.ts]: Done

File: src/routes/(user)/credits/+page.server.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { redirect } from '@sveltejs/kit';
+import { getUserCredits, getUserCreditTransactions } from '$lib/utils/supabase/admin';
+
+export async function load({ locals: { safeGetSession } }) {
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/routes/(user)/credits/+page.server.ts]: Done

File: src/routes/(user)/credits/+page.svelte [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+		margin: 0 1rem;
+		padding: 0.5rem 1rem;
+		border: 1px solid #ccc;
+		border-radius: 0.5rem;
+		cursor: pointer;
+	}
+	button.font-bold {
+		font-weight: bold;
+		background-color: #f0f0f0;
+	}
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/routes/(user)/credits/+page.svelte:162]: Create an enhancement ticket for this in the 2.0.0 milestone unassigned

**Comment by jcreek** [comment]
@OllyNicholass this PR is now ready for review :)

**Review by OllyNicholass — CHANGES_REQUESTED** [review]
Can we add a link to the credits page.
Nice work

**Review by OllyNicholass — APPROVED** [review]

**Pull request merged**